### PR TITLE
feat: Get signals working with sendAppMessage

### PR DESCRIPTION
**Pull request by jamsea**

### CODE REVIEW

**Review by lazeratops — COMMENTED** [review]
Looking great, just a couple of tiny things and some questions for my own education.

**Comment by lazeratops** [comment]
Just a reminder that our public facing demo app is using the deployed shims, maybe good to include it in the testing flow for these shims PRs: https://github.com/daily-demos/opentok-shim-conversion-demo

File: src/session/DailyEventHandler.ts [hunk 1]
```diff
@@ -27,6 +27,36 @@ export class DailyEventHandler {
...
+    const d = dailyEvent.data;
+
+    const connection: OT.Connection = {
+      connectionId: dailyEvent.fromId,
+      creationTime: new Date().getTime(),
+      data: "",
+    };
+
+    const signalEvent: Event<"signal", Session> & {
+      type?: string;
```
[the remark this answers is not in the document]
Comment (jamsea) [reply src/session/DailyEventHandler.ts:44]: I double-checked `opentok.d.ts`. Their signal type looks like this:

```typescript
    signal: Event<'signal', Session> & {
      type?: string;
      data?: string;
      from: Connection | null;
    };
```

There's probably a better way to give event emitters types in TypeScript. I haven't quite figured out how 🤔 Even our DailyJs library event types are closer to a 9/10 than a 10/10.

I'm not sure if it's worth it to make it into a reusable public interface for two locations. I'm not sure where the best place to put this interface would be. Maybe I can create a function called `createSignalEvent()` like you did for the other events that may solve this. I'll poke around a bit.

File: src/session/DailyEventHandler.ts [hunk 2]
```diff
@@ -27,6 +27,36 @@ export class DailyEventHandler {
...
+    } = {
+      // @ts-expect-error - this is a mistake in the OpenTok types https://tokbox.com/developer/guides/signaling/js/
+      type: d.type,
+      data: d.data,
+      from: connection,
+    };
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (signalEvent.type) {
+      this.ee.emit(`signal:${signalEvent.type}`, signalEvent);
```
[the remark this answers is not in the document]
Comment (jamsea) [reply src/session/DailyEventHandler.ts:56]: I just confirmed we need both by checking original Opentok code: 
image
image

[the remark this answers is not in the document]
Comment (lazeratops) [reply src/session/DailyEventHandler.ts:56]: Great, thanks for checking!

**Review by lazeratops — APPROVED** [review]
LGTM once CI is happy!

**Pull request merged**